TypeTreeDiffGUI/UnityVersionComparer.cs

```diff
@@ -0,0 +1,366 @@
+		public bool IsGreater(int major, int minor, int build, UnityVersionType type, int typeNumber) => this > new UnityVersion(major, minor, build, type, typeNumber);
+		public bool IsGreater(string version) => this > Parse(version);
+
+		public bool IsGreaterEqual(int major) => this >= From(major);
+		public bool IsGreaterEqual(int major, int minor) => this >= From(major, minor);
+		public bool IsGreaterEqual(int major, int minor, int build) => this >= From(major, minor, build);
+		public bool IsGreaterEqual(int major, int minor, int build, UnityVersionType type) => this >= From(major, minor, build, type);
+		public bool IsGreaterEqual(int major, int minor, int build, UnityVersionType type, int typeNumber) => this >= new UnityVersion(major, minor, build, type, typeNumber);
+		public bool IsGreaterEqual(string version) => this >= Parse(version);
+		#endregion
+
+		#region From
+		private UnityVersion From(int major)
+		{
+			ulong data = ((ulong)(major & 0xFFFF) << 48) | (0x0000FFFFFFFFFFFFUL & m_data);
+			return new UnityVersion(data);
+		}
+		private UnityVersion From(int major, int minor)
+		{
+			ulong data = ((ulong)(major & 0xFFFF) << 48) | ((ulong)(minor & 0xFF) << 40) | (0x000000FFFFFFFFFFUL & m_data);
+			return new UnityVersion(data);
+		}
+		private UnityVersion From(int major, int minor, int build)
+		{
+			ulong data = ((ulong)(major & 0xFFFF) << 48) | ((ulong)(minor & 0xFF) << 40) | ((ulong)(build & 0xFF) << 32) |
+				(0x00000000FFFFFFFFUL & m_data);
+			return new UnityVersion(data);
+		}
+		private UnityVersion From(int major, int minor, int build, UnityVersionType type)
+		{
+			ulong data = ((ulong)(major & 0xFFFF) << 48) | ((ulong)(minor & 0xFF) << 40) | ((ulong)(build & 0xFF) << 32) |
+				((ulong)((int)type & 0xFF) << 24) | (0x0000000000FFFFFFUL & m_data);
+			return new UnityVersion(data);
+		}
+		private UnityVersion From(int major, int minor, int build, UnityVersionType type, int typeNumber)
+		{
+			ulong data = ((ulong)(major & 0xFFFF) << 48) | ((ulong)(minor & 0xFF) << 40) | ((ulong)(build & 0xFF) << 32)
+				| ((ulong)((int)type & 0xFF) << 24) | ((ulong)(typeNumber & 0xFF) << 16) | (0x000000000000FFFFUL & m_data);
+			return new UnityVersion(data);
+		}
+		#endregion
+	}
+
+    public static class Extensions
+    {
+	    public static int ParseDigit(this char _this)
+	    {
+		    return _this - '0';
+	    }
+
+	    public static string ToLiteral(this UnityVersionType _this)
+	    {
+		    return _this switch
+		    {
+			    UnityVersionType.Alpha => "a",
+			    UnityVersionType.Beta => "b",
+			    UnityVersionType.China => "c",
+			    UnityVersionType.Final => "f",
+			    UnityVersionType.Patch => "p",
+			    UnityVersionType.Experimental => "x",
+			    _ => throw new Exception($"Unsupported vertion type {_this}"),
+		    };
+	    }
+    }
+
+    public enum UnityVersionType
+    {
+	    Alpha = 0,
+	    Beta,
+	    China,
+	    Final,
+	    Patch,
+	    Experimental,
+
+	    MaxValue = Experimental,
+    }
+}
```

Comment:
Use the nuget package

Reply:
There is a package for that 👀 What is it called?